applications/sim_showers_for_trigger_rates.py

```diff
@@ -0,0 +1,186 @@
+
+    Example
+    -------
+    Producing a set of proton showers for trigger rate simulations of LST.
+
+    .. code-block:: console
+
+        python applications/sim_showers_for_trigger_rates.py -a 4LST -s North \
+        --primary proton --nruns 100 --nevents 10000 --output {some dir for large files}
+'''
+
+import logging
+import argparse
+
+import astropy.units as u
+
+import simtools.io_handler as io
+import simtools.config as cfg
+import simtools.util.general as gen
+from simtools.shower_simulator import ShowerSimulator
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+        description=(
+            'Calculate and plot the PSF and eff. mirror area as a function of off-axis angle '
+            'of the telescope requested.'
+        )
+    )
+    parser.add_argument(
+        '-a',
+        '--array',
+        help='Name of the array (e.g. 1MST, 4 LST ...)',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '-s',
+        '--site',
+        help='Site name (North or South)',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '--primary',
+        help='Name of the primary particle (e.g. proton, helium ...)',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '--nruns',
+        help='Number of runs (default=100)',
+        type=int,
+        default=100
+    )
+    parser.add_argument(
+        '--nevents',
+        help='Number of events/run (default=100)',
+        type=int,
+        default=100000
+    )
+    parser.add_argument(
+        '--zenith',
+        help='Zenith angle in deg (default=20)',
+        type=float,
+        default=20
+    )
+    parser.add_argument(
+        '--azimuth',
+        help='Azimuth angle in deg (default=0)',
+        type=float,
+        default=0
+    )
+    parser.add_argument(
+        '--output',
+        help='Path of the output directory where the simulations will be saved.',
+        type=str,
+        default=None
+    )
+    parser.add_argument(
+        '--test',
+        help='Test option will not submit any job.',
+        action='store_true'
+    )
+    parser.add_argument(
+        '-v',
+        '--verbosity',
+        dest='logLevel',
+        action='store',
+        default='info',
+        help='Log level to print (default is INFO)'
+    )
+
+    args = parser.parse_args()
+    label = 'trigger_rates'
+
+    logger = logging.getLogger()
+    logger.setLevel(gen.getLogLevelFromUser(args.logLevel))
+
+    # Output directory to save files related directly to this app
+    outputDir = io.getApplicationOutputDirectory(cfg.get('outputLocation'), label)
+
+    showerConfigData = {
+        'corsikaDataDirectory': args.output,
+        'site': args.site,
+        'layoutName': args.array,
+        'runRange': [1, args.nruns + 1],
+        'nshow': args.nevents,
+        'primary': args.primary,
+        'erange': [100 * u.GeV, 300 * u.TeV],
```

Comment:
100 GeV as lower energy?

Reply:
That is wrong, indeed. I will change to 3 GeV by now. However, different particles have different lower energy limit in CORSIKA. We have to figure out a way to take this into account.
We could have a safety system in corsika_config that changes the energy automatically in case the user has set a invalid value.

Example: the user set 1 GeV for helium but the minimum allowed is 10 GeV. Then we automatically change it to 10 GeV.

A warning should be printed, of course. But still, one could argue that is not a good idea to change the user setting on the background like this. What do you think?

Reply:
Why is the minimum for helium set to 10 GeV?

Otherwise printing out a warning is fine.

Reply:
Not sure I agree that changing the user's setting and just printing a warning is fine. I think we should never change an explicit user setting and continue running, if we think a setting is wrong we should always stop running (or allow a force mode of running).